src/sparseml/transformers/utils/helpers.py

```diff
@@ -224,3 +211,106 @@ def resolve_sequence_length(config: AutoConfig) -> int:
         "(inferred from HF transformers config) "
     )
     return sequence_length
+
+
+def resolve_recipe_application(
+    recipe: Union[str, Path, None], model_path: Union[str, Path]
+) -> Union[str, Path, None]:
+    """
+    Resolve the recipe to apply to the model.
+    :param recipe: the recipe to apply to the model.
+        It can be one of the following:
+        - None
+        - a path to the recipe file
+        - name of the recipe file (e.g. "recipe.yaml")
+            (assumed to be stored in the model_path instead
+            of RECIPE_NAME)
+        - a string containing the recipe
+
+        If recipe is None, the function will try to
+        find the recipe in the model_path if model_path
+        is a local path. If model_path is a huggingface
+        model_id, the function will try to download the
+        recipe from huggingface.co.
+
+    :param model_path: the path to the model to load
+    :return: the resolved recipe
+    """
+
+    if recipe is None:
+        # if recipe is None -> still look for recipe.yaml in the model_path
```

Comment:
I think this case could be cleaned up into it's own function, something like if recipe is None: infer_recipe_from_model_path(...), which would return path to a local recipe (either already present in model_path)
or downloaded from SparseZoo/Huggingface

Reply:
This would clean up the function as well? which is a little difficult to follow as of now?

Reply:
sure!